### PR TITLE
Homepage Pagination

### PR DESCRIPTION
**Pull request by tommypoa**
### Feature Description
- Functionality of pagination present in home page of the web. 
- Number of works per page is hardcoded to 6 to visualize functionality based on the number of seed works we have
- Styling is not present

### Related PRs
#84 

### Migrations
None.

### Tests Performed, Edge Cases
Tested on home page

### Screenshots
sfai pagination demo
CC: @by-co

### CODE REVIEW

**Comment by by-co** [comment]
Won't merge until next week, when we should have styling done :)

File: app/javascript/components/artists/ArtistProfile.jsx [hunk 1]
```diff
@@ -68,6 +67,23 @@ class ArtistProfile extends React.Component {
...
+  updateFeatured = (work_id) => {
+    let formData = new FormData();
+    formData.append(`artist[featured_work_id]`, work_id);
+    fetch(APIRoutes.artists.update(this.props.artist.id), {
+      method: 'PUT',
+      body: formData,
+      credentials: 'same-origin',
+      headers: {
+        "X_CSRF-Token": document.getElementsByName("csrf-token")[0].content
+      }
```
[the remark this answers is not in the document]
Comment (by-co) [reply app/javascript/components/artists/ArtistProfile.jsx]: @lbkchen It's merged in #74 (discussion [here](https://cal-blueprint.slack.com/archives/CGH67HNRL/p1551749554009300) abt why GH is weird) — good call out though!

**Comment by tommypoa** [comment]
@by-co Yes! I will get them in by the end of the week

**Comment by by-co** [comment]
@tommypoa @emilyzhong Awesome teamwork here! It's beautiful :)

Screen Shot 2019-03-18 at 7 47 18 PM
Screen Shot 2019-03-18 at 7 47 31 PM

File: app/javascript/components/artists/AllArtists.jsx [hunk 2]
```diff
@@ -81,14 +98,30 @@ class AllArtists extends React.Component {
...
+            <h1>Artists</h1>
+            <nav className="li-indigo pagination" role="navigation" aria-label="Pagination Navigation">
+              <ReactPaginate
+              previousLabel={"\u00ab"}
+              nextLabel="&raquo;"
+              breakLabel={'...'}
+              breakClassName={'break-me'}
+              pageCount={this.state.pageCount}
+              marginPagesDisplayed={2}
+              pageRangeDisplayed={5}
```
Comment (by-co) [inline app/javascript/components/artists/AllArtists.jsx]: Hmm this probably shouldn't be hardcoded, it's displaying empty pages bc we don't have a ton of seeded artists

Comment (tommypoa) [reply app/javascript/components/artists/AllArtists.jsx]: Hmm the displaying of empty pages in AllArtists page should not be happening + isn't happening when I'm loading it locally. **marginPagesDisplayed=2** should make it such that if there are more than 4 pages, the first 2 and last 2 pages will be displayed [1,2, 19 and 20 in the pic], and for **pageRangeDisplayed=5** refer to the middle 5 pages [5, 6, 7, 8, 9 in the pic], if there are enough pages. 

Think it should be hardcoded based on we want the pagination bar to look, or if it should be a dynamic value then I can implement that as well!

image

Comment (by-co) [reply app/javascript/components/artists/AllArtists.jsx]: Screen Shot 2019-03-19 at 2 02 03 PM

Not sure why I'm seeing so many pages ):

Comment (by-co) [reply app/javascript/components/artists/AllArtists.jsx]: Looks like we'll want to use `artists.filter(artist => artist.works.length > 0)` to determine the          `pageCount`

File: app/javascript/components/artists/AllArtists.jsx [hunk 3]
```diff
@@ -11,7 +14,10 @@ class AllArtists extends React.Component {
       artists: null,
       filters: [],
       componentDidMount: false,
-      isLoading: true
+      isLoading: true,
+      pageCount: 0,
+      workStartIndex: 0,
```
Comment (by-co) [inline app/javascript/components/artists/AllArtists.jsx]: Nit:
```suggestion
      artistStartIndex: 0,
```

**Pull request merged**